script.js

```diff
@@ -0,0 +1,160 @@
+
+  // Create branch if not present
+  if (!branchExists) {
+    const ref = await octokit
+      .request("POST /repos/{owner}/{repo}/git/refs", {
+        owner,
+        repo,
+        ref: `refs/heads/${branchName}`,
+        sha,
+      })
+      .then((response) => response.data.ref);
+
+    if (!ref) {
+      octokit.log.warn(`Error creating branch in ${repository.html_url}`);
+
+      return;
+    }
+  }
+
+  // Get all files from .github/workflows folder
+  const { data: files } = await octokit.request(
+    "GET /repos/{owner}/{repo}/contents/{path}",
+    {
+      owner,
+      repo,
+      path,
+    }
+  );
+
+  if (!Array.isArray(files)) {
+    throw new Error("The path is not a folder"); //TODO
+  }
+
+  const workflowFiles = files.filter((file) => isYamlFile(file.name));
+  let workflowsUpdated = false;
+
+  for (const workflowFile of workflowFiles) {
+    const { data, updated } = await composeCreateOrUpdateTextFile(octokit, {
+      owner,
+      repo,
+      path: workflowFile.path,
+      branch: branchName,
+      message: `ci(workflow): add '${cache}' cache for actions/setup-node in ${workflowFile.name}`,
+      content: ({ exists, content }) => {
+        const yamlDocument = parseDocument(content);
+        const jobs = yamlDocument.get('jobs')
+
+        for(let i = 0; i < jobs.items.length; i++) {
+          const job = jobs.items[i].value
+          const steps = job.get('steps')
+          for(let j = 0; j <steps.items.length; j++) {
+            const step = steps.get(j)
+            const stepUses = step.get('uses')
+            const stepWith = step.get('with')
+            
+            if (
+              stepUses &&
+              stepUses.includes("actions/setup-node") &&
+              !stepWith.get('cache')
+            ) {
+              stepWith.set('cache', cache);
+            }
+          }
+        };
+
+        return prettier.format(
+          yamlDocument.toString(),
+          {
+            parser: "yaml",
+          }
+        );
+      },
+    });
+
+    if (updated) {
+      octokit.log.info(
+        `${path} updated in ${repository.html_url} via ${data.commit.html_url}`
+      );
+
+      workflowsUpdated = true;
+    }
+  }
+
+  if (workflowsUpdated) {
+    //
+    // Pull Request
+    //
+
+    // Create pull request
+    const { data: pr } = await octokit.request(
+      "POST /repos/{owner}/{repo}/pulls",
+      {
+        owner,
+        repo,
+        head: branchName,
+        base: defaultBranch,
+        title: "ci(workflow): add cache to workflows using actions/setup-node",
+      }
+    );
+
+    octokit.log.info(`Create Pull Request at ${pr.html_url}`);
+
+    // Add the "maintenance" label to the pull request
+    await octokit.request(
+      "POST /repos/{owner}/{repo}/issues/{issue_number}/labels",
+      {
+        owner,
+        repo,
+        issue_number: pr.number,
+        labels: ["maintenance"],
+      }
+    );
```

Comment:
This is @octokit specific, I wouldn't force the `maintenance` label on other users